lib/src/supabase_auth.dart

```diff
@@ -21,12 +27,23 @@ class SupabaseAuth {
   /// {@macro supabase.localstorage.accessToken}
   Future<String?> get accessToken => _localStorage.accessToken();
 
+  /// Returns when the initial session recovery is done.
+  /// Can be used to determine whether a user is signed in or not uplon
+  /// initial app launch.
+  Future<Session?> get initialSession => _initialSessionCompleter.future;
+  final Completer<Session?> _initialSessionCompleter = Completer();
```

Comment:
Added a completer that will complete upon initial session recovery with a session if it existed. 

Developers can await for `SupabaseAuth.instance.initialSession` to get the initial auth state of the signed in user. What do all of you think of this new variable?

Reply:
I like that name. But I would prefer returning a wrapper around `Session` to be able to return an error. Currently, you can't defer between a failed recovery and a missing persisted session.

Reply:
@Vinzent03 Great point! Let me change it to `Completer<GotrueSessionResponse>` right now!

Reply:
@Vinzent03 

I actually ended up not going with `Completer<GotrueSessionResponse>`, but keeping the type as is and using `completeError` when error occured. 

@bdlukaa has been working on amazing PRs that will throw error instead of returning them when error occurs, so wanted to align with it.

Reply:
Right, I remember the other ongoing reworks. Great.